Cue/app/api/CueApi.js

```diff
@@ -21,33 +34,36 @@ class CueApi {
 				'X-CUE-ACCESS-TOKEN': this.accessToken,
 			}
 		}
+		this.setNetworkActivityIndicatorVisible(true)
 		return fetch(serverURL + endpoint, {
 			method: method,
 			headers: headers,
 			body: body,
 		})
 		.then(checkStatus)
-		.then(response => response.text())
+		.then(response => response.text())
 		.then(text => {
+			this.setNetworkActivityIndicatorVisible(false)
 			try {
 				return JSON.parse(text)
 			} catch (e) {
 				return {}
 			}
 		})
 		.catch(e => {
+			this.setNetworkActivityIndicatorVisible(false)
```

Comment:
WOW this is a sign I have been looking at too much code. Not sure how I missed that `try`-`catch` block :confused: